src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java

```diff
@@ -99,12 +100,58 @@ protected String name() {
             return "Property Mutator";
         }
 
-        private final Caller mMutator;
-        private final Caller mAccessor;
+        private final PropertyCaller mMutator;
+        private final PropertyCaller mAccessor;
         private final WeakHashMap<View, Object> mOriginalValues;
         private final Object[] mOriginalValueHolder;
     }
 
+    public static class LayoutSetVisitor extends ViewVisitor {
+        public LayoutSetVisitor(List<Pathfinder.PathElement> path, LayoutCaller mutator) {
+            super(path);
+            mOriginalValues = new WeakHashMap<View, int[]>();
+            mMutator = mutator;
+        }
+
+        @Override
+        public void cleanup() {
+            for (Map.Entry<View, int[]> original:mOriginalValues.entrySet()) {
+                final View changedView = original.getKey();
+                final int[] originalValue = original.getValue();
+                mMutator.applyMethodWithArguments(changedView, originalValue);
```

Comment:
cleanup() should probably explicitly set layout rules rather than call something indirectly.